pkg/e2e/autoscaler/autoscaler.go

```diff
@@ -207,40 +218,24 @@ var _ = g.Describe("[Feature:Machines] Autoscaler should", func() {
 		})
 		o.Expect(err).NotTo(o.HaveOccurred())
 
-		workLoad := newWorkLoad()
-
 		// We want to clean up these objects on any subsequent error.
 		defer func() {
-			if workLoad != nil {
-				cascadeDelete := metav1.DeletePropagationForeground
-				wait.PollImmediate(1*time.Second, e2e.WaitShort, func() (bool, error) {
-					if err := client.Delete(context.TODO(), workLoad, func(opt *runtimeclient.DeleteOptions) {
-						opt.PropagationPolicy = &cascadeDelete
-					}); err != nil {
-						glog.Errorf("error querying api for workLoad object: %v, retrying...", err)
-						return false, nil
-					}
-					return true, nil
-				})
-				glog.Info("Deleted workload object")
+			err = e2e.DeleteObjectsByLabels(context.TODO(), client, map[string]string{autoscalingTestLabel: ""}, &batchv1.JobList{})
+			if err != nil {
+				glog.Warning(err)
 			}
+			glog.Info("Deleted workload object")
 
-			wait.PollImmediate(1*time.Second, e2e.WaitShort, func() (bool, error) {
-				if err := client.Delete(context.TODO(), machineAutoscaler); err != nil {
-					glog.Errorf("error querying api for machineAutoscaler object: %v, retrying...", err)
-					return false, nil
-				}
-				return true, nil
-			})
+			err = e2e.DeleteObjectsByLabels(context.TODO(), client, map[string]string{autoscalingTestLabel: ""}, &caov1alpha1.MachineAutoscalerList{})
+			if err != nil {
+				glog.Warning(err)
```

Comment:
It is OK to continue in the face of an error?

Reply:
It's in the defer part where all the created resources are deleted. So, if this one fails, there are still other resources to be deleted.

Reply:
Maybe good to add:

```Go
// if this one fails, there are still other resources to be deleted.
```

Reply:
done